### DISCUSSION TITLE
Debian 12 forbids pip install (known problem)- therefore, based on below, FastAPI cannot be installed

### DISCUSSION THREAD

**Discussion by dgl-dev** [fastapi/fastapi#12206] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
## pip install failure on Debian 12
~~~
 pip install fastapi
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.11/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification. pip install fastapi
error: externally-managed-environment
~~~

## FastAPI Failure after install in conda venv
~~~
(cenv-fastapi) limawhiskeydelta@penguin:~/anaconda3/envs/cenv-fastapi$ fastapi dev proj1/main.py
To use the fastapi command, please install "fastapi[standard]":

        pip install "fastapi[standard]"

Traceback (most recent call last):
  File "/home/limawhiskeydelta/anaconda3/envs/cenv-fastapi/bin/fastapi", line 11, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/limawhiskeydelta/anaconda3/envs/cenv-fastapi/lib/python3.12/site-packages/fastapi/cli.py", line 12, in main
    raise RuntimeError(message)  # noqa: B904
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: To use the fastapi command, please install "fastapi[standard]":

        pip install "fastapi[standard]"
~~~
```

### Description

1. Create conde venv
2. Conda install fastAPI
3. Create dir proj1
4. make main.py - copy from FastAPI docs
5. make ___init__.py
6. execute fastapi dev proj1/main.py
7. get failure requiring pip install fastapi

### Operating System

Linux

### Operating System Details

Debian 12

### FastAPI Version

pkgs/main/linux-64::fastapi-0.112.2-py312h06a4308_0

### Pydantic Version

N/A

### Python Version

Python 3.12.5

**Comment by AIdjis** [comment, marked as answer]
This issue occurred because you're trying to install packages system-wide using pip, but your Python environment is managed by the system's package manager (such as apt in Debian-based systems).
- To fix this, create a virtual environment and install FastAPI through it:
```console
python3 -m venv myenv
source myenv/bin/activate
pip install fastapi
```
